src/magic_folder/config.py

```diff
@@ -862,6 +855,20 @@ def store_local_snapshot(self, cursor, snapshot):
 
         :param LocalSnapshot snapshot: The snapshot to store.
         """
+        for parent in snapshot.parents_local:
+            cursor.execute(
+                """
+                SELECT
+                    count(*) from [local_snapshots]
+                WHERE
+                    identifier= ?
+                """,
+                (unicode(parent.identifier),),
+            )
+            count = cursor.fetchone()
+            if count[0] != 1:
+                raise LocalSnapshotMissingParent(parent.identifier)
```

Comment:
I think this check may end up being temporary, as I think it may make sense to move the logic from [here](https://github.com/LeastAuthority/magic-folder/blob/a53f0bab97dfa897ad535ae9f23c56b78cb52429/src/magic_folder/uploader.py#L121-L138) into the call to `store_local_snapshots` below (as part of #461).

Reply:
^ would probably be better as a comment in the code describing why the check is there (and under what conditions you think it can go away).

Reply:
I don't think it makes sense to have a comment in the code about removing the check. I think it is information that is useful to a reviewer, as I am adding a check in one PR, but already have a WIP PR (#471) where I then turn around and review it.

I don't think I think it makes sense to add a comment to code, about how a potential interface change to this code, would remove the need for the check. The check makes sense as the code is, but obviously doesn't with the anticipated change. In particular, as it is *potential* change[1], it seems like a comment is something that we easily get out of date.

[1] In writing this comment, I realized that plan I had doesn't handle one particular case, and so the plan I had for changing this may not actually make sense, which strengthens my feeling that it doesn't make sense to mention it in the code.

---

Here is a bit of an explanation of the changes I was intending:

The WIP code I have changes this function to take a local snapshot with unspecified parents, so I replace this check with one that checks that that both `.parents_local` and `.parents_raw` are `None` (for unspecified) as opposed to a (possibly empty) list. I do this so we can find the current snapshot (which will be the parent) in the same transaction we write the data for it. I wanted to do this so that if a local parent snapshot is uploaded and becomes remote as we are writing the stash file, we aren't left with a dangling snapshot identifier).

However, I realized that (in some cases), we could also race with the downloader in a way that would cause the calculated parent to become incoherent, so I made need to take a different approach. I added some further discussion of my current thoughts on this race [here](https://github.com/tp-la/magic-folder/commit/5d69e7c4def99d3024b67eef6c5a81090bacdf48), which is currently part of #471.

Reply:
> I don't think it makes sense to have a comment in the code about ...

I guess different things might make different amounts of sense to different people.  In the case of this kind of documentation request, maybe we should err on the side of explaining more things in more places, hoping that anyone who goes looking for an explanation will find at least _something_ that helps them out?

> In particular, as it is potential change[1], it seems like a comment is something that we easily get out of date.

I'm not sure what it would get out of date with.  Is the plan written down somewhere else?  If so, we don't have to be redundant - adding a link in the code to wherever else that is still does a better job of maximizing the chances of everyone understanding (as compared to not adding any explanation).

Reply:
> In the case of this kind of documentation request, maybe we should err on the side of explaining more things in more places

While I generally agree with this, I don't think it applies in this case. It was added to forestall a question from a reviewer about why one PR added a check, and then the next removed it. I think it is only relevant in the context of reviewing this and #471 in reasonably close succesion, which, in hindsight hasn't happened. 

In particular, I think adding a comment about this to the code makes it *less* clear.